apv5sdk-v15/apps/int6000-utils-1.1.6/ram/sdram.7
.TH sdram 7 "Intellon Corporation, Ocala FL USA" "int6000-utils-linux" "Intellon Linux Toolkit"
.SH NAME
sdram \- Intellon SDRAM Configuration File Generator
.SH SYNOPSIS
\fBsdram\fR 
.SH DESCRIPTION
The Intellon SDRAM Configuration File Generator creates two INT6300 SDRAM configuration files in the current working folder. Either of them may be used as the argument for the \fBint6kf\fR \fB-C\fR option when initializing INT6300 devices. Read the \fBint6kf\fR man page for more information.
.PP
This program is part of the Intellon Linux Toolkit. See the \fBilt\fR man page for an overview and installation instructions.
.SH BACKGROUND
Configuration files for the Linux Toolkit have \fB.cfg\fR extensions and consist of 32 binary bytes plus a 32 bit checksum. They are difficult to modify but easy to verify. The format requires no conversion on input or output and, consequently, it is preferred to that used by the Windows Device Manager.
.PP
Configuration files for the Windows Device Manager have \fB.config\fR extensions and consist of 64 hexadecimal ASCII characters. They are easily modified using a text editor but cannot be easily validated. They require ASCII to binary conversion on input and binary to ASCII conversion on output. Read the \fBconfig2cfg\fR man page for more information.
.SH OPTIONS
.TP
There are no options.
.SH EXAMPLES
The following example writes two files, \fBsdram16mb.cfg\fR and \fBsdram64.cfg\fR, in the current working folder. They will replace any existing files having the same names.
.PP
    # sdram
    writing sdram16mb.cfg
    writing sdram64mb.cfg
.SH DISCLAIMER
The SDRAM file structure and content is proprietary to Intellon Corporation, Ocala FL USA. Consequently, public information is not available. Intellon Corporation reserves the right to change the file structure or content in future firmware releases without any obligation to notify or compensate users of this program.
.SH SEE ALSO
.BR config2cfg ( 7 ),
.BR int6kf ( 7 )
.SH CREDITS
 Charles Maier <charles.maier@intellon.com>


apv5sdk-v15/apps/int6000-utils-1.1.6/ram/sdram.c
/*====================================================================*
 *   
 *   Copyright (c) 2010, Atheros Communications Inc.
 *   
 *   Permission to use, copy, modify, and/or distribute this software 
 *   for any purpose with or without fee is hereby granted, provided 
 *   that the above copyright notice and this permission notice appear 
 *   in all copies.
 *   
 *   THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL 
 *   WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED 
 *   WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL  
 *   THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR 
 *   CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM 
 *   LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, 
 *   NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN 
 *   CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *   
 *--------------------------------------------------------------------*/

/*====================================================================*
 *
 *   sdram.c - create INT6300 sdram configuration files;
 *
 *   sdram.h
 *
 *   this program writes two sdram configuration files in the current
 *   working directory; both files are for Linux toolkit programs and
 *   not the Windows Device Manager;
 *
 *   one of these sdram configuration blocks must be sent to the 
 *   INT6300 bootloader with VS_SET_SDRAM before downloading NVM 
 *   and PIB files;
 *
 *   This software and documentation is the property of Intellon 
 *   Corporation, Ocala, Florida. It is provided 'as is' without 
 *   expressed or implied warranty of any kind to anyone for any 
 *   reason. Intellon assumes no responsibility or liability for 
 *   errors or omissions in the software or documentation and 
 *   reserves the right to make changes without notification. 
 *   
 *   Intellon customers may modify and distribute the software 
 *   without obligation to Intellon. Since use of this software 
 *   is optional, users shall bear sole responsibility and 
 *   liability for any consequences of it's use. 
 *
 *.  Intellon INT6x00 Linux Toolkit for HomePlug AV; 
 *:  Published 2007 by Intellon Corp. ALL RIGHTS RESERVED;
 *;  For demonstration; Not for production use;
 *
 *   Contributor(s):
 *	Charles Maier <charles.maier@intellon.com>
 *
 *--------------------------------------------------------------------*/

#include <stdio.h>
#include <unistd.h>
#include <stdint.h>
#include <fcntl.h>

#include "../tools/memory.h"
#include "../ram/sdram.h"

#ifndef MAKEFILE
#include "../tools/checksum_32.c"
#endif

#define FILE1 "sdram16mb.cfg"
#define FILE2 "sdram64mb.cfg"

int main (int argc, const char * argv []) 

{
	const uint8_t sdram16mb [32] = 
	{
		0x00,
		0x00,
		0x00,
		0x01,
		0x68,
		0x2f,
		0x14,
		0x00,
		0x92,
		0xd4,
		0xe1,
		0x01,
		0xd6,
		0x83,
		0x08,
		0x00,
		0x88,
		0x32,
		0x00,
		0x00,
		0xdb,
		0x06,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00 
	};
	const uint8_t sdram64mb [32] = 
	{
		0x00,
		0x00,
		0x00,
		0x04,
		0x88,
		0x31,
		0x14,
		0x00,
		0x91,
		0xd4,
		0xe1,
		0x01,
		0xe3,
		0x2b,
		0x01,
		0x00,
		0x89,
		0x30,
		0x00,
		0x00,
		0x66,
		0x03,
		0x00,
		0x00,
		0x00,
		0x01,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00,
		0x00 
	};
	int fd;
	uint32_t checksum;
	if ((fd = open (FILE1, O_CREAT|O_WRONLY|O_TRUNC, 0444)) != -1) 
	{
		printf ("writing %s\n", FILE1);
		checksum = checksum_32 (sdram16mb, sizeof (sdram16mb), 0);
		write (fd, sdram16mb, sizeof (sdram16mb));
		write (fd, &checksum, sizeof (checksum));
		close (fd);
	}
	if ((fd = open (FILE2, O_CREAT|O_WRONLY|O_TRUNC, 0444)) != -1) 
	{
		printf ("writing %s\n", FILE2);
		checksum = checksum_32 (sdram64mb, sizeof (sdram64mb), 0);
		write (fd, sdram64mb, sizeof (sdram64mb));
		write (fd, &checksum, sizeof (checksum));
		close (fd);
	}
	return (0);
}

